whitepapers/Tables.md

```diff
@@ -88,28 +91,59 @@ Each row corresponds to a Transformer implementation in python. (TODO: describe
+| `raw_datum_id`          |    int    |    ✓    |          |  ✓  |                 | RawDatum to be transformed                                                                                    |
+| `ingest_config_id`      |    int    |    ✓    |          |  ✓  |                 | IngestConfig used                                                                                             |
+| `status`                | enum(int) |    ✓    |          |     |     CREATED     | Status of the transformer run, one of {CREATED, STARTED, SUCCEEDED, FAILED, RESCHEDULED}                      |
+| `error`                 |   text    |         |          |     |       ""        | A custom error message or traceback describing why this job failed                                            |
+| `completions`           |    int    |         |          |     |        0        | The number of times `status` has been set to SUCCEEDED                                                        |
+| `date_started`          | datetime  |         |    ✓     |     |                 | Datetime `status` was last set to STARTED                                                                     |
+| `date_created`          | datetime  |         |          |     |       now       | Datetime this row was created                                                                                 |
+| `date_modified`         | datetime  |         |          |     | now (on update) | Datetime this row was last modified                                                                           |
+| `share_version`         |   text    |         |          |     |     UNKNOWN     | The commitish at the time this job was last run                                                               |
+| `transformer_version`   |   text    |    ✓    |          |     |   000.000.000   | Semantic version of the transformer, with each segment padded to 3 digits (e.g. '1.2.10' => '001.002.010')    |
+| `regulator_version`     |   text    |    ✓    |          |     |   000.000.000   | Semantic version of the regulator, with each segment padded to 3 digits (e.g. '1.2.10' => '001.002.010')      |
+| `consolidator_version`  |   text    |    ✓    |          |     |   000.000.000   | Semantic version of the consolidator, with each segment padded to 3 digits (e.g. '1.2.10' => '001.002.010')   |
+| `ingest_config_version` |   text    |    ✓    |          |     |   000.000.000   | Semantic version of the `IngestConfig`, with each segment padded to 3 digits (e.g. '1.2.10' => '001.002.010') |
 
 #### Other indices
 * `raw_id`, `transformer_version` (unique)
+
+#### Notes
+* `regulator_version` and `consolidator_version` will be mutable. Whenever the regulator or consolidator version gets bumped existing jobs should be updated.
```

Comment:
Ideally, when the queue isn't that large, the Janitor should queue up a couple of these tasks.
That way whenever we make changes we can just sit back and wait for the Janitor to make sure that everything is updated.